File: app/src/main/java/com/yonder/exercise/shared/listeners/ViewPagerOnPageChangeListener.java
package com.yonder.exercise.shared.listeners;

import android.support.v4.view.ViewPager;

/**
 * Created by YusufMac on 01/06/17.
 */

public abstract class ViewPagerOnPageChangeListener implements ViewPager.OnPageChangeListener {
    @Override
    public void onPageScrolled(int i, float v, int i1) {
    }

    @Override
    public void onPageSelected(int i) {
        onPageChanged(i);
    }

    @Override
    public void onPageScrollStateChanged(int i) {
    }

    public abstract void onPageChanged(int position);
}

File: app/src/main/java/com/yonder/exercise/shared/listeners/ViewPagerOnPageChangeListener.kt
package com.yonder.exercise.shared.listeners

import android.support.v4.view.ViewPager

/**
 * Created by YusufMac on 01/06/17.
 */

abstract class ViewPagerOnPageChangeListener : ViewPager.OnPageChangeListener {
    override fun onPageScrolled(i: Int, v: Float, i1: Int) {}

    override fun onPageSelected(i: Int) {
        onPageChanged(i)
    }

    override fun onPageScrollStateChanged(i: Int) {}

    abstract fun onPageChanged(position: Int)
}
